core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1426,7 +1426,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * supported for Hadoop-supported filesystems.
    */
   def addFile(path: String, recursive: Boolean): Unit = {
-    val uri = new URI(path)
+    val uri = new Path(path).toUri
```

Comment:
Meanwhile, we could alternatively use `Utils.resolveURI` as well which we are already testing within  Spark. However, this util seems not hadling `C:/a/b/c` case (not `C:\a\b\c` which we should fix). So I suggested `Path (...).toUri` instead but if you feel strongly about this, we could use that. I will try to find and share doc and tests for `Path` as I got in my home though.